### PR TITLE
 Document global `fallbacks_for_empty_translations` setting 

### PR DESCRIPTION
**Pull request by parndt**
This is based on the commits in PR #467, extended by @phansch :smile: 

Closes #467
Closes phansch/globalize#1

### CODE REVIEW

**Comment by phansch** [comment]
Oh woops, looks like I opened the PR on my fork instead of here :man_facepalming: 

It looks like some tests around the fallbacks are failing now. I'll take a look later today and see what these failures are about.

**Comment by parndt** [comment]
Thanks @phansch

**Comment by phansch** [comment]
The tests seem to pass only when not using the refactorings from 1b5ae4393. It seems, this commit changed behaviour of globalize, because

1) `fallbacks_for` starts returning false for `nil` inputs
2) `[true, false].include? value` doesn't coerce as `!!value != value`, which changed behaviour in some places, too

I pushed a revert of the commit to see if CI passes.

File: lib/globalize.rb [hunk 1]
```diff
@@ -54,7 +54,7 @@ def default_fallbacks(for_locale = self.locale)
     end
     
     def fallbacks_for_empty_translations=(value = false)
-      return if [true, false].include? value
+      return if !!value != value # Only allow booleans
```
Comment (parndt) [inline lib/globalize.rb]: Curious.. how does this improve on `[true, false].include?(value)`? That version only allows booleans too. :smile:

Comment (parndt) [reply lib/globalize.rb]: Or do you mean this restricts what is passed to the method as `value`?

Comment (parndt) [reply lib/globalize.rb]: if that's our intention, perhaps we should just guard:
```ruby
raise ArgumentError, 'fallbacks_for_empty_translations= can only be true or false' if [true, false].exclude?(value)
```

**Comment by phansch** [comment]
Slipped off my radar, I will try and get to this next week :smile:

**Comment by parndt** [comment]
Thanks @phansch

**Pull request closed without merging**